### PR TITLE
Changed recommendation for testing library

### PR DESCRIPTION
**Pull request by NataliaTepluhina**
## Description of Problem

As described in https://github.com/vuejs/docs/issues/2295, vue-testing-library has issued with testing async components with Suspense. We cannot recommend it over Vue Test Utils that deal with Suspense fine

## Proposed Solution

Recommend VTU over testing library

## Additional Information

Close https://github.com/vuejs/docs/issues/2295

### CODE REVIEW

File: src/guide/scaling-up/testing.md [hunk 1]
```diff
@@ -186,9 +186,13 @@ mount(Stepper, {
...
-cy.get(valueSelector).should('be.visible').and('contain.text', '0')
-  .get(buttonSelector).click()
-  .get(valueSelector).should('contain.text', '1')
+cy.get(valueSelector)
+  .should('be.visible')
+  .and('contain.text', '0')
+  .get(buttonSelector)
+  .click()
+  .get(valueSelector)
+  .should('contain.text', '1')
```
Comment (skirtles-code) [inline src/guide/scaling-up/testing.md:195]: Are these formatting changes intentional?

File: src/guide/scaling-up/testing.md [hunk 2]
```diff
@@ -207,21 +211,21 @@ cy.get(valueSelector).should('be.visible').and('contain.text', '0')
...
-
 - [`@vue/test-utils`](https://github.com/vuejs/test-utils) is the official low-level component testing library that was written to provide users access to Vue specific APIs. It's also the lower-level library `@testing-library/vue` is built on top of.
 
-We recommend using `@testing-library/vue` for testing components in applications, as its focus aligns better with the testing priorities of applications. Use `@vue/test-utils` only if you are building advanced components that require testing Vue-specific internals.
+- [`@testing-library/vue`](https://github.com/testing-library/vue-testing-library) is a Vue testing library focused on testing components without relying on implementation details. Built with accessibility in mind, its approach also makes refactoring a breeze. Its guiding principle is that the more tests resemble the way software is used, the more confidence they can provide.
+
+We recommend using `@vue/test-utils` for testing components in applications, as its focus aligns better with the testing priorities of applications. `@testing-library/vue` has issues with testing asynchronous component with Suspense, so it should be used with caution.
```
Comment (skirtles-code) [inline src/guide/scaling-up/testing.md:228]: The bit about how 'its focus aligns better with the testing priorities of applications' seemed specific to `@testing-library/vue`. Does it still make sense to have that comparison when switching the recommendation?

**Review by skirtles-code — COMMENTED** [review]
I don't have much personal experience of working with either library, so I'm not really able to judge what the recommendation should be.

As Suspense is still considered experimental, I'm not sure whether `@testing-library/vue` should be expected to have proper support for it.

File: src/guide/scaling-up/testing.md (hunk 1 above)
Comment (NataliaTepluhina) [reply src/guide/scaling-up/testing.md]: No, it was autoformatter. Removed the changes.

File: src/guide/scaling-up/testing.md (hunk 2 above)
Comment (NataliaTepluhina) [reply src/guide/scaling-up/testing.md]: Makes sense, I've removed this bit

**Comment by NataliaTepluhina** [comment]
@skirtles-code thank you for the review! I've addressed your comments, could you please take another look?

**Comment by pyoho-agn** [comment]
@NataliaTepluhina It may make sense to update the `Recipe` section at the bottom of this document with a `@vue/test-utils` example instead of a `testing-library` one. It's a bit confusing that there is a recommendation to use `@vue/test-utils`, but then have the largest example not use it. Apologies if this should be a new issue.

**Pull request merged**